src/games/supported/DemonAttack.cpp

```diff
@@ -121,3 +129,41 @@ void DemonAttackSettings::loadState(Deserializer & ser) {
   m_lives = ser.getInt();
 }
 
+// returns a list of mode that the game can be played in
+ModeVect DemonAttackSettings::getAvailableModes(){
+    ModeVect modes;
+    modes.push_back(1);
+    modes.push_back(3);
+    modes.push_back(5);
+    modes.push_back(7);
+
+    return modes;
+}
+
+// set the mode of the game
+// the given mode must be one returned by the previous function
+void DemonAttackSettings::setMode(game_mode_t m, System &system, StellaEnvironment& environment){
+    if(m==1 || m==3 || m==5 || m==7){
```

Comment:
I agree that a pythonic syntax would be nice here. However the c++ equivalent :
` if(std::find(getAvailableModes().begin(), getAvailableModes().end(), m) != getAvailableModes().end()) `
doesn't look that pretty, does it ?